### PR TITLE
chore: cicd, CKAN 2.9.x

### PR DESCRIPTION
**Pull request by duttonw**
Get a release out that supports CKAN 2.9.x prior to dropping support.

https://github.com/ckan/ckanext-validation/pull/103 will look after the uplift (as well as incorporating parts of qld-gov-au downstream fork where needed)

### CODE REVIEW

**Comment by ThrawnCA** [comment]
Tests are failing because of compatibility problems with `frictionless` wanting a newer version of `jinja`. (Which is one of the obstacles that has prevented the qld-gov-au fork from becoming compatible with upstream).

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+      if: ${{ matrix.ckan-version == 2.9 }}
+      run: pip install "setuptools>=44.1.0,<71"
+
+    - name: Install requirements
+      run: |
+        pip install -r requirements.txt
+        pip install -r dev-requirements.txt
+        echo "no-warn-conflicts install of jinja2 and markupsafe"
+        pip install --no-warn-conflicts jinja2==2.10.1
+        pip install --no-warn-conflicts markupsafe==2.0.1
```
Comment (ThrawnCA) [inline .github/workflows/publish.yml:101]: This could be combined with the previous line.

**Review by ThrawnCA — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+          - ckan-version: "2.9"
+            ckan-image: "2.9-py3.9"
+          #- ckan-version: "master" Publish does not care about master
+          #  ckan-image: "master"
+      fail-fast: false
+
+    name: CKAN ${{ matrix.ckan-version }}
+    runs-on: ubuntu-latest
+    container:
+      image: openknowledge/ckan-dev:${{ matrix.ckan-version }}
```
[the remark this answers is not in the document]
Comment (duttonw) [reply .github/workflows/publish.yml:65]: i'm getting a 2.9 cut to pypi then we will be moving to ckan/ckan-dev which has test failure.

File: .github/workflows/publish.yml (hunk 1 above)
Comment (duttonw) [reply .github/workflows/publish.yml:101]: this is how it was previously, just leaving it alone until we do the 2.10+ uplift.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -1,38 +1,56 @@
...
-        run: flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics --exclude ckan
+        run: flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics --extend-exclude ckan
 
   test:
     needs: lint
     strategy:
       matrix:
-        ckan-version: [2.9]
+        include: #ckan-image see https://github.com/ckan/ckan-docker-base, ckan-version controls other image tags
+#          - ckan-version: "2.11"
```
Comment (ThrawnCA) [inline .github/workflows/test.yml:24]: Why not uncomment these since they are marked experimental?

**Review by ThrawnCA — APPROVED** [review]

**Pull request merged**